internal/store/postgres/billing_transactions_repository.go

```diff
@@ -388,9 +422,28 @@ func (r BillingTransactionRepository) getBalanceInTx(ctx context.Context, tx *sq
 // in transaction table till now.
 func (r BillingTransactionRepository) GetBalance(ctx context.Context, accountID string) (int64, error) {
 	var amount int64
-	if err := r.dbc.WithTxn(ctx, sql.TxOptions{}, func(tx *sqlx.Tx) error {
+	if err := r.dbc.WithTxn(ctx, sql.TxOptions{
+		Isolation: sql.LevelSerializable,
```

Comment:
This is done to prevent race condition?

Reply:
I feel it will make system slow once we start to have good amount of transactions. Also we calculate balance by summing over all rows instead which may also get performance issues.

Reply:
> Is this done to prevent race condition?

Yes, this is to prevent race conditions. What other way do you propose here to prevent it?

> I feel it will make system slow once we start to have good amount of transactions. Also we calculate balance by summing over all rows instead which may also get performance issues.

Yes, one way I was thinking was to maintain a monthly statement in system and only accumulate records since last statement. But this introduces complexity quite a lot and was postponing it until we hit the performance degradation. I want to know what part of it is the bottleneck before going ahead with this approach. Do you have anything else in mind?